File: 5-Problem-A-Day/209. Minimum Size Subarray Sum.java
class Solution {
    public int minSubArrayLen(int s, int[] nums) {
        int len = nums.length + 1, start = 0, sum = 0;
        for (int i = 0; i < nums.length; i++) {
            sum += nums[i];
            while (sum >= s) {
                len = Math.min(len, i - start + 1);
                sum -= nums[start];
                start++;
            }
        }
        return len == nums.length + 1 ? 0 : len;
    }
}

File: 2023/209. Minimum Size Subarray Sum.cpp
class Solution {
public:
    int minSubArrayLen(int target, vector<int>& nums) {
        int start = 0, minLen = nums.size() + 1, sum = 0, index = 0;
        while (index < nums.size()) {
            sum += nums[index];
            while (start < index && sum - nums[start] >= target) {
                sum -= nums[start];
                start++;
            }
            if (sum >= target) {
                minLen = min(minLen, index - start + 1);
            }
            index++;
        }
        return minLen > nums.size() ? 0 : minLen;
    }
};